model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/hibernate/jsonb/JsonbType.java

```diff
@@ -115,6 +116,13 @@ abstract class IgnoredMetadataFieldsMixIn {
         @JsonIgnore public abstract Boolean isClientRole();
     }
 
+    abstract static class EnumsMixIn implements EnumWithUnchangeableIndex {
+
+        // we convert enums to its index and vice versa
+        @Override
+        @JsonValue public abstract Integer getUnchangeableIndex();
```

Comment:
How exactly is the conversion from the integer back into the enum value performed here? If I get this right, when writing the object into a JSON string, we will get something like `"enumProperty": "2"`. What I am missing here is when the JSON string is read again into an object, shouldn't some of the `getValueOfIntger` methods be called?

Reply:
Ah, that makes sense now. Thank you!